src/main/java/com/aws/greengrass/logmanager/model/ProcessingFiles.java

```diff
@@ -35,25 +39,72 @@ public ProcessingFiles(int maxInactiveTimeSeconds) {
         this.maxInactiveTimeSeconds = maxInactiveTimeSeconds;
     }
 
+    /**
+     * Stores the currently processing file information only if the entry does not exist.
+     *
+     * @param value - currently processing file information
+     */
+    public void putIfAbsent(LogManagerService.CurrentProcessingFileInformation value) {
+        if (cache.get(value.getFileHash()) == null) {
```

Comment:
use computeIfAbsent

Reply:
that's because you're calling put, don't call put from here.

Reply:
Yeah I know, but on put I am also setting the mostRecentlyUsed which is something that I would still want. Instead of doing it in to places I just do it in put and have this just delegate.